application/src/test/java/be/jpendel/domain/event/EventFactoryTest.java

```diff
@@ -0,0 +1,24 @@
+package be.jpendel.domain.event;
+
+import be.jpendel.application.CreateEventCommand;
+import org.junit.Test;
+
+import java.time.LocalDateTime;
+
+public class EventFactoryTest {
+
+    @Test
+    public void eventWithSameNameAndSameStartdateLocationCanNotBeCreated(){
+       EventFactory factory = new EventFactory();
+        LocalDateTime time = LocalDateTime.now();
+        CreateEventCommand eventCmd = CreateEventCommand.builder()
+                .withEndDateTime(time).withLocation("loc")
+                .withStartDateTime(time)
+                .withName("test").build();
+        Event e1 = factory.create(eventCmd);
+        EventRepository repo = new InMemEventRepository();
+        repo.add(e1);
+        repo.add(e1);// moet dit falen ?
```

Comment:
Het contract dat u repo aangaat ten opzichte van het domain moet wel geïmplementeerd worden door een InMemory repo eh. Als het contract in de prod code dicteert dat het moet falen op een unique constraint, dan moet u inMemory versie dat ook doen. Het is niet omdat we ervoor kiezen dat dit soort dingen een runtime ex. geeft dat dat geen onderdeel is van het contract van u repository.